File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.py
def solution(survey, choices):
    answer = ''
    result = {
        "R": 0, "T": 0,
        "C": 0, "F": 0,
        "J": 0, "M": 0,
        "A": 0, "N": 0,
    }

    for v,c in zip(survey, choices):
        if c < 4: result[v[0]] += (4 - c)
        elif c > 4: result[v[1]] += (c - 4)
        else: continue
        
    types = list(result.keys())
    
    for l,r in zip(types[::2], types[1::2]):
        if result[l] < result[r]: answer += r
        else: answer += l
    
    return answer

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.js
function solution(survey, choices) {
    let result = {
        "R": 0, "T": 0,
        "C": 0, "F": 0,
        "J": 0, "M": 0,
        "A": 0, "N": 0,
    };
    
    const types = ["RT","CF","JM","AN"];

    for(let i = 0; i < survey.length; i++){
        result[choices[i]<4 ? survey[i][0] : survey[i][1]] += Math.abs(choices[i] - 4);
    }

    return types.map(([a, b]) => result[b] > result[a]? b : a).join("");
}